lib/matplotlib/axes/_axes.py

```diff
@@ -6409,23 +6409,32 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
              orientation='vertical', rwidth=None, log=False,
              color=None, label=None, stacked=False, **kwargs):
         """
-        Plot a histogram.
+        Compute and plot a histogram.
 
-        Compute and draw the histogram of *x*.  The return value is a tuple
-        (*n*, *bins*, *patches*) or ([*n0*, *n1*, ...], *bins*, [*patches0*,
-        *patches1*, ...]) if the input contains multiple data.  See the
-        documentation of the *weights* parameter to draw a histogram of
-        already-binned data.
+        This method uses `numpy.histogram` to bin the data in *x* and count the
+        number of values in each bin, then draws the distribution either as a
+        `.BarContainer` or `.Polygon`. The *bins*, *range*, *density*, and
+        *weights* parameters are forwarded to `numpy.histogram`. If the data
```

Comment:
This reads great now! I suggest to make a new paragraph here for the plot-binned-data topic.

```suggestion
        *weights* parameters are forwarded to `numpy.histogram`.
        
        If the data
```